### PR TITLE
Various perf and size improvements

### PR DESCRIPTION
**Pull request by JoviDeCroock**
- shorter functions for easy optimization
- getting rid of rendererState
- make mount and patch component more inline

### CODE REVIEW

File: src/component.js [hunk 1]
```diff
@@ -12,9 +12,7 @@ export let ENABLE_CLASSES = false;
  * @type {import('./internal').RendererState}
  */
 export const rendererState = {
```
Comment (JoviDeCroock) [inline src/component.js:14]: With something like https://github.com/preactjs/preact/pull/3493 in mind the state could still be a good option @developit

**Comment by JoviDeCroock** [comment]
https://github.com/preactjs/preact/pull/3704/commits/4e31d427804b9758bc2ae42fe8334cdc1c0ccbf0 seems to make `filter_list` ever so slightly slower as opposed to keeping it on the `renderState` I think this could be due to traversing the context upwards on a regular basis with patches

**Review by marvinhagemeister — APPROVED** [review]
This is great! Find it a little bit easier to read too without the property access stuff  💯

**Review by developit — APPROVED** [review]
Hell yeah

**Pull request merged**